sklearn/utils/_seq_dataset.pyx.tp

```diff
@@ -314,26 +321,25 @@ cdef class CSRDataset{{name_suffix}}(SequentialDataset{{name_suffix}}):
             The weight of each sample.
         """
         # keep a reference to the data to prevent garbage collection
-        self.X_data = X_data
-        self.X_indptr = X_indptr
-        self.X_indices = X_indices
-        self.Y = Y
-        self.sample_weights = sample_weights
+        self.X_data = X_data.base
+        self.X_indptr = X_indptr.base
+        self.X_indices = X_indices.base
+        self.Y = Y.base
+        self.sample_weights = sample_weights.base
```

Comment:
Attributes of `SequentialDataset{{name_suffix}}` and its subclasses are declared in `sklearn/utils/_seq_dataset.pxd.tp` and uses `cnp.ndarray`.

They must be changed to use memoryview instead and be `const`-qualified if possible.

Also do not that `*_ptr` might be made irrelevant by such changes and might be removed in another PR if this is the case.